docs/proposals/scheduler/agent.proto

```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+
+message Empty {}
+
+
+// ------------------
+// Lifecycle
+// ------------------
+
+message ConnectionRequest {
+    string uuid = 1;
+}
+
+message ConnectionResponse {
+    enum StatusCode {
+        GRANTED = 0;
+        DENIED = 1;
+    }
+
+    StatusCode status_code = 1;
+}
+
+service LifecycleService {
+    rpc Connect(ConnectionRequest) returns (ConnectionResponse);
+    rpc Disconnect(Empty) returns (Empty);
+}
+
+
+// ------------------
+// Status updates
+// ------------------
+
+message NodeStatus {
+    message Memory {
+        uint64 total = 1;
+        uint64 free = 2;
+    }
+
+    message CpuLoad {
+        double load = 1;
+    }
+
+    Memory memory = 1;
+    CpuLoad cpu_load = 2;
+}
+
+message InstanceStatus {
+    enum StatusCode {
+        WAITING = 0;
+        RUNNING = 1;
+        TERMINATED = 2;
+    }
+
+    message Resources {
+        int32 cpu = 1;
+        int32 memory = 2;
+        int32 disk = 3;
+    }
+
+    string name = 1;
+    StatusCode status_code = 2;
+    Resources resource_usage = 3;
+    string message = 4;
+}
+
+message InstanceStatuses {
+    repeated InstanceStatus instance_statuses = 1;
+}
+
+service StatusUpdateService {
+    rpc UpdateNodeStatus(NodeStatus) returns (Empty);
```

Comment:
Here you want thee agent to call into that interface, with a stream of node statuses. It is better than getting periodic syncs, as you will always be out of sync between 2 such synchronization points.